test/tests/fallback.json

```diff
@@ -4,52 +4,59 @@
   "description": "Test cases for fallback behaviour.",
   "defaultTestProperties": {
     "bidiIsolation": "none",
-    "locale": "en-US",
-    "expErrors": true
+    "locale": "en-US"
   },
   "tests": [
     {
       "description": "function with unquoted literal operand",
       "src": "{42 :test:function fails=format}",
       "exp": "{|42|}",
-      "expParts": [{ "type": "fallback", "source": "|42|" }]
+      "expParts": [{ "type": "fallback", "source": "|42|" }],
+      "expErrors": [{ "type": "bad-operand" }]
```

Comment:
```suggestion
      "expErrors": [{ "type": "bad-option" }]
```

Reply:
Hmm... I found this confusing until I remembered that we agreed to use `bad-option` to avoid creating a label for `message-function-error` in the **_not at all official_** definition under tests [here](https://github.com/unicode-org/message-format-wg/tree/main/test#error-codes). Non-WG members won't have the benefit of having been on the telecon when we discussed it 😉 

Unlike the other classes of error, MFE can have implementation-defined errors. We should make this clearer in the test suite text so that implementations do not merely ape our selection of `bad-option` when an internal exception/error occurs. I think it would be okay to commit this, but we need to put more effort into the text in the /tests page to make super clear that `bad-option` is not required by non-test functions.

Reply:
Agreed on all counts. I also noticed while implementing this change that we lack a generic error for failures in formatting or selection, which could be due to the operand, options, or the phase of the moon. As in, values for which resolution succeeds, but then the `MessageValue.formatToString()` or similar call throws an error. It's pretty likely to involve some bad option, but that's not actually guaranteed -- and beyond the scope of this PR.